### PR TITLE
Better builtin support in ResourceReloader

### PR DESCRIPTION
**Pull request by RSlysz**
### Purpose of this PR
Fix regression introduced by PR https://github.com/Unity-Technologies/Graphics/pull/6169 
Issue is that Resources.GetBuiltinResource raise error if the path is not right instead of returning null as expected in the implementation of previous PR.

Also now raize expression when atempting to use [ReloadGroup] on ScriptableObject

---
### Testing status
Manual tested the exception raized for ScriptableObject with [ReloadGroup]

---
### Comments to reviewers
@FrancescoC-unity With this fix you should be able to still use search of fbx in builtin resource normally.

@phi-lira this is an alternative proper fix to https://github.com/Unity-Technologies/Graphics/pull/7066/files as the code make possible to look at buildin resource when not using builtin path which make no sens.

### CODE REVIEW

File: com.unity.render-pipelines.core/Runtime/Utilities/ResourceReloader.cs [hunk 1]
```diff
@@ -190,19 +190,23 @@ static UnityEngine.Object Load(string path, Type type, bool builtin)
...
+                {
+                    result = Resources.GetBuiltinResource(type, path);
+
+                    if (IsNull(result))
+                        result = AssetDatabase.GetBuiltinExtraResource(type, path);
+                }
             }
             else
             {
                 result = AssetDatabase.LoadAssetAtPath(path, type);
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.core/Runtime/Utilities/ResourceReloader.cs:209]:
```suggestion
      result = (type == typeof(Shader)) ? Shader.Find(path) : AssetDatabase.LoadAssetAtPath(path, type);
```

I wonder if we should use the `Shader.Find` here too.

**Review by alex-vazquez — APPROVED** [review]
Looks good to me

**Review by phi-lira — APPROVED** [review]

**Review by alex-vazquez — APPROVED** [review]
Looks good.

**Review by FrancescoC-unity — APPROVED** [review]

**Comment by RSlysz** [comment]
CI issue found:
- RenderPipelineManagerCallbackTests.BeginAndEndCameraRenderingCallbackMatch_Camera test fail, seems unrelated
- RenderPipelineManagerCallbackTests.BeginAndEndCameraRenderingCallbackMatch_CustomRender test fail, seems unrelated
- RenderPipelineManagerCallbackTests.BeginAndEndCameraRenderingCallbackMatch_FullscreenPassthrough test fail, seems unrelated
- RenderPipelineManagerCallbackTests.BeginAndEndCameraRenderingCallbackMatch_RenderToTexture test fail, seems unrelated
- UnityEngine.Rendering.Utils.Tests.TimedScopeTests.SimpleTimeCheckFromPtr test fail (solved here https://github.com/Unity-Technologies/Graphics/pull/7088)
- UnityEngine.Rendering.Utils.Tests.TimedScopeTests.SimpleTimeCheckFromRef test fail (solved here https://github.com/Unity-Technologies/Graphics/pull/7088)
- UniversalGraphicsTests.202_SSAO_Depth test fail, seems unrelated
- some metal error (seems unrelated)
- some timeout (hdrp-dx): known issue, unrelated
- some issue with android (seems to be a bokken issue)

No more `Failed to find <path>`. Seems this fix works.

**Pull request merged**